compiler/rustc_passes/src/stability.rs

```diff
@@ -816,24 +816,54 @@ impl<'tcx> Visitor<'tcx> for Checker<'tcx> {
                         }
                     }
 
-                    // `#![feature(const_trait_impl)]` is unstable, so any impl declared stable
-                    // needs to have an error emitted.
-                    if features.const_trait_impl()
-                        && self.tcx.is_const_trait_impl(item.owner_id.to_def_id())
-                        && const_stab.is_some_and(|(stab, _)| stab.is_const_stable())
-                    {
-                        self.tcx.dcx().emit_err(errors::TraitImplConstStable { span: item.span });
+                    if features.const_trait_impl() && hir::Constness::Const == *constness {
```

Comment:
nit: inconsistency between `==` here but `if let` below for matching constness. can you choose one?